synapse/handlers/profile.py

```diff
@@ -455,7 +455,11 @@ async def _update_remote_profile_cache(self) -> None:
                 continue
 
             new_name = profile.get("displayname")
+            if new_name is not None and not isinstance(new_name, str):
+                new_name = None
             new_avatar = profile.get("avatar_url")
+            if new_avatar is not None and not isinstance(new_avatar, str):
+                new_avatar = None
```

Comment:
I think all we need is the `isinstance` check here?

```suggestion
            if not isinstance(new_name, str):
                new_name = None
            new_avatar = profile.get("avatar_url")
            if not isinstance(new_avatar, str):
                new_avatar = None
```